### ISSUE TITLE
[Bug] Lacking materials for object indicator/feature

### ISSUE THREAD

**Issue by BraedonWooding** [TeamPorcupine/ProjectPorcupine#1557]
### Expected behavior
- Shouldn't allow user to 'schedule' an object to be created if you are lacking the resources
- If for some reason an object can't be completed due to lacking resources it should be indicated with a 'red' tint or similar.
- It should indicate the resources required for each object

### Actual behavior
- No indicator/s and can queue up objects regardless of resources.

### Specs
- User can't create a job for an object if they are lacking the resources to construct it.
- A red tint or similar on any job that can't be completed (and a tooltip explaining why; 'can't be reached', 'lacking materials/resources')
- A construct tooltip indicating resources x/y with x being the amount of that resource required and y being your total amount of that resource.

**Comment by koosemose** [comment]
I disagree on the first point, user should be able to queue up as many items to be created as they want, to be completed when materials become available.

**Comment by BraedonWooding** [comment]
Okay, sure; the more important part was focusing on saying that it is lacking resources?

**Comment by BraedonWooding** [comment]
![image](https://cloud.githubusercontent.com/assets/22880786/19505140/c4befbd8-960b-11e6-9235-e97b907e260e.png)

That's just throwing together an example?  Of what it could look like (colours were just randomly chosen).  And that text would only appear if you moused over the '!'.

**Comment by koosemose** [comment]
I think just changing the tint from green to red would be good enough to show a problem, mouseover info would be of use (perhaps using our mouseover info window, rather than a popup sort of design)

**Comment by BraedonWooding** [comment]
mmm, idk still doesn't feel significant enough like a colour tint is easy to miss

**Comment by koosemose** [comment]
![image](https://cloud.githubusercontent.com/assets/6334643/19528739/2e2de400-95fb-11e6-9b6b-73c9c2d12056.png)
Doesn't seem that hard to miss to me. And if we're warning them when they place something, they're less likely to not already know so we don't want to make it to hugely attention grabbing. Perhaps a bit of a flashing like effect if resources were available but then become non-available, though I'm not sure how necessary that itself is.

**Comment by Geoffrotism** [comment]
the red seems like enough to show that something is preventing this from being built. then a hover over it could have a tooltip explaining why. 
- "Need x y." where x is the number needed, and y is the material. ex: "need 5 iron plates"
- "Unreachable Job"

etc.

**Comment by BraedonWooding** [comment]
Okay well I will make a PR of this after making the changes to the other one
